File: app/src/main/java/com/example/meyss/javaretrofitdagger/data/PokAttacks.java
package com.example.meyss.javaretrofitdagger.data;

import java.util.List;

import androidx.room.Embedded;
import androidx.room.Relation;

public class PokAttacks {
    @Embedded
    public Pokemon pok;
    @Relation(
            parentColumn = "id",
            entityColumn = "pokId"
               )
    public List<Attack> attacks;

    @Override
    public String toString() {
        return "PokAttacks{" +
                "pok=" + pok +
                ", attacks=" + attacks +
                '}';
    }
}

File: app/src/main/java/com/example/meyss/javaretrofitdagger/data/PokAttacks.kt
package com.example.meyss.javaretrofitdagger.data

import androidx.room.Embedded
import androidx.room.Relation

class PokAttacks {
    @Embedded
    var pok: Pokemon? = null
    @Relation(parentColumn = "id", entityColumn = "pokId")
    var attacks: List<Attack>? = null

    override fun toString(): String {
        return "PokAttacks{" +
                "pok=" + pok +
                ", attacks=" + attacks +
                '}'
    }
}
